test/Date/toStringAndToUTCStringYearPadding.js

```diff
@@ -0,0 +1,56 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+// For dates whose years are less than 1000, Date.toString() and Date.toUTCString() should pad the years
+// to 4 digits.
+// See https://github.com/Microsoft/ChakraCore/pull/4067
+
+runTest(new Date("0001-10-13T05:16:33Z"),
+    "Fri Oct 12 0001 22:16:33 GMT-0700 (Pacific Daylight Time)",
+    "Sat, 13 Oct 0001 05:16:33 GMT");
+
+runTest(new Date("0011-10-13T05:16:33Z"),
+    "Wed Oct 12 0011 22:16:33 GMT-0700 (Pacific Daylight Time)",
+    "Thu, 13 Oct 0011 05:16:33 GMT");
+
+runTest(new Date("0111-10-13T05:16:33Z"),
+    "Mon Oct 12 0111 22:16:33 GMT-0700 (Pacific Daylight Time)",
+    "Tue, 13 Oct 0111 05:16:33 GMT");
+
+runTest(new Date("1111-10-13T05:16:33Z"),
+    "Thu Oct 12 1111 22:16:33 GMT-0700 (Pacific Daylight Time)",
+    "Fri, 13 Oct 1111 05:16:33 GMT");
+
+// Negative years are also padded to four digits
+runTest(new Date("-000001-10-13T05:16:33Z"),
+    "Tue Oct 12 -0001 22:16:33 GMT-0700 (Pacific Daylight Time)",
+    "Wed, 13 Oct 0002 B.C. 05:16:33 GMT");

```

Comment:
I wonder why `toUTCString` not only prints GMT timezone (as opposed to UTC), but also switches from negative year to B.C. year. I think we should consult the spec and other engines on what the outputs should look like.

Reply:
For this date (`-000001-10-13T05:16:33Z`), both v8 and SpiderMonkey output GMT timezone and negative years: `Wed, 13 Oct -001 05:16:33 GMT` and `Wed, 13 Oct -0001 05:16:33 GMT` respectively.